### PR TITLE
refactor(esl-utils): make ScrollUtils functional, update API

### PR DESCRIPTION
**Pull request by Natalie-Smirnova**
Closes: #1039

### CODE REVIEW

**Review by ala-n — APPROVED** [review]

**Comment by ala-n** [comment]
@Natalie-Smirnova thanks for reworking this API
Just a two-note that needs to be discussed in the nearest weekly:
- it looks like we also need a test for such functionality as well. Going to create a ticket
- it also needs to be discussed if we need to synchronize these updates with the update regarding scroll lock (mentioned as a JS one, but basically it is mostly about an ability to lock container scrolls)

**Review by NastaLeo — APPROVED** [review]

**Review by fshovchko — APPROVED** [review]

File: src/modules/esl-utils/dom/scroll/utils.ts [hunk 1]
```diff
@@ -3,52 +3,60 @@ export type ScrollStrategy = 'none' | 'native' | 'pseudo';
...
-    initiator && initiatorSet.add(initiator);
-    (initiatorSet.size > 0) && ScrollUtils.lock(strategy);
+/**
+ * Disable scroll on the page.
+ * @param strategy - to make scroll visually disabled
+ * @param initiator - object to associate request with
+ *
+ * TODO: currently requests with different strategy is not taken into account
+ * */
+export function lockScroll(strategy?: ScrollStrategy | null, initiator?: any): void {
```
Comment (ala-n) [inline src/modules/esl-utils/dom/scroll/utils.ts:32]: Draft:
```ts
export function lockScroll(
  target: Element = document.documentElement, 
  options?: {
     strategy?: ScrollStrategy | null, // support for page only for now
     initiator?: any // Lock/unlock only wia the same marker
  }
): void {
```

File: src/modules/esl-utils/dom/scroll/utils.ts [hunk 2]
```diff
@@ -3,52 +3,60 @@ export type ScrollStrategy = 'none' | 'native' | 'pseudo';
...
+
+/** @deprecated use functional utils instead */
+export abstract class ScrollUtils {
+  public static readonly hasVerticalScroll = hasVerticalScroll;
+  public static readonly hasHorizontalScroll = hasHorizontalScroll;
+
+  public static readonly lock = lockScroll;
+  public static readonly unlock = unlockScroll;
+  public static readonly requestLock = (initiator: any, strategy?: ScrollStrategy | null) => lockScroll(strategy, initiator);
+  public static readonly requestUnlock = (initiator: any) => unlockScroll(initiator);
```
Comment (ala-n) [inline src/modules/esl-utils/dom/scroll/utils.ts:61]: As a result of weekly discussio: remove, mark as a breaking change for 4.0.0

**Comment by ala-n** [comment]
Included into: #1051 with the core team discussion update

**Pull request closed without merging**